sdk/storage/azure-storage-file-datalake/azure/storage/file/datalake/_data_lake_directory_client.py

```diff
@@ -0,0 +1,512 @@
+from ._models import DirectoryProperties
+from ._path_client import PathClient
+
+
+class DataLakeDirectoryClient(PathClient):
+    def __init__(
+        self, account_url,  # type: str
+        file_system_name,  # type: str
+        directory_name,  # type: str
+        credential=None,  # type: Optional[Any]
+        **kwargs  # type: Any
+    ):
+        super(DataLakeDirectoryClient, self).__init__(account_url, file_system_name, directory_name,
+                                                      credential=credential, **kwargs)
+
+    def generate_shared_access_signature(self):
+        # ???
+        pass
+
+    @classmethod
+    def from_connection_string(
+            cls, conn_str,  # type: str
+            file_system_name,  # type: str
+            directory_name,  # type: str
+            credential=None,  # type: Optional[Any]
+            **kwargs  # type: Any
+        ):  # type: (...) -> DataLakeDirectoryClient
+        """
+        Create DataLakeDirectoryClient from a Connection String.
+
+        :param str conn_str:
+            A connection string to an Azure Storage account.
+        :param file_system_name: The name of file system to interact with.
+        :type file_system_name: str
+        :param directory_name: The name of directory to interact with. The directory is under file system.
+        :type directory_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, and account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+            Credentials provided here will take precedence over those in the connection string.
+        :return a DataLakeDirectoryClient
+        :rtype ~azure.storage.file.datalake.DataLakeDirectoryClient
+        """
+        account_url, secondary, credential = parse_connection_str(conn_str, credential, 'dfs')
+        if 'secondary_hostname' not in kwargs:
+            kwargs['secondary_hostname'] = secondary
+        return cls(
+            account_url, file_system_name=file_system_name, directory_name=directory_name,
+            credential=credential, **kwargs)
+
+    def create_directory(self, content_settings=None,  # type: Optional[ContentSettings]
+                         metadata=None,  # type: Optional[Dict[str, str]]
+                         **kwargs):
+        # type: (...) -> Dict[str, Union[str, datetime]]
+        """
+        Create directory
+
+        :param ~azure.storage.file.datalake.ContentSettings content_settings:
+            ContentSettings object used to set path properties.
+        :param metadata:
+            Name-value pairs associated with the blob as metadata.
+        :type metadata: dict(str, str)
+        :keyword ~azure.storage.file.datalake.DataLakeLeaseClient or str lease:
```

Comment:
There's a bug in sphinx that causes the keyword type to render incorrectly in the docs when done like this. See this [issue](https://github.com/Azure/azure-sdk-for-python/issues/8132). A workaround is to use paramtype like this:
`:paramtype: ~azure.storage.file.datalake.DataLakeLeaseClient or str`